### PR TITLE
Offline support json serializer for do

### PR DESCRIPTION
**Pull request by jaskooo**

### CODE REVIEW

File: syncano-ios/SCConstants.m [hunk 1]
```diff
@@ -95,12 +95,10 @@ + (NSString *)DB_PATH {
 }
 
 + (NSString *)createTableSQLStatement {
-    NSString *objectsTableSchema = @"CREATE TABLE IF NOT EXISTS %@ ("
-    @"Syncano TEXT PRIMARY KEY, "
+    NSString *objectsTableSchema = @"CREATE TABLE IF NOT EXISTS SyncanoDataObjects ("
```
Comment (lifcio) [inline syncano-ios/SCConstants.m]: Is this the only place where you use this table name? I think previous approach made sense, with using a variable here, that would hold table name inside.

File: syncano-ios/SCDataObject+LocalStorage.m [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+#pragma mark - Saving -
+- (void)saveToLocalStorageWithCompletion:(SCCompletionBlock)completion {
+    [[Syncano localStore] saveDataObject:self withCompletionBlock:completion];
+}
+
+#pragma mark - Helpers -
+- (void)generateInsertQueryWithCompletion:(void(^)(NSError *error, NSString* query))completion {
+    NSError *error;
+    NSString *insertQuery = [NSString stringWithFormat:@"INSERT INTO SyncanoDataObjects (className, objectId,json) VALUES ('%@',%@,'%@');",NSStringFromClass([self class]),self.objectId,[[SCParseManager sharedSCParseManager] JSONSerializedDictionaryFromDataObject:self error:&error]];
```
Comment (lifcio) [inline syncano-ios/SCDataObject+LocalStorage.m]: I see we use table name explicitly here as well. If we ever changed it, we would need to be sure to change it everywhere. Better to store i as a variable.

File: syncano-ios/SCLocalStore.m [hunk 3]
```diff
@@ -32,7 +33,19 @@ - (void)initializeDB {
 }
 
 - (void)saveDataObject:(SCDataObject *)dataObject withCompletionBlock:(SCCompletionBlock)completionBlock {
-    
+    if ([_db open]) {
```
Comment (lifcio) [inline syncano-ios/SCLocalStore.m]: Why not use `self.db` here?

File: syncano-ios/SCLocalStore.m [hunk 4]
```diff
@@ -32,7 +33,19 @@ - (void)initializeDB {
...
+                [_db beginTransaction];
+                [_db executeUpdate:query];
+                [_db  commit];
+            }
+            [_db close];
+            if (completionBlock) {
+                completionBlock(error);
+            }
+        }];
+    }
```
Comment (lifcio) [inline syncano-ios/SCLocalStore.m]: What happens if `[_db open]` returns false? Completion block will never be called - probably should still be called, just with an error.

File: syncano-ios/Syncano.m [hunk 5]
```diff
@@ -12,13 +12,13 @@
 #import "SCAPIClient+Class.h"
 #import "SCLocalStore.h"
 
+static SCLocalStore *localStore;
```
Comment (lifcio) [inline syncano-ios/Syncano.m]: Since it's an internal variable, maybe use underscore at the beginning, e.g. `__localStore`.

**Comment by lifcio** [comment]
Why moved all `SCClassRegisterItem` to main class? Not saying it's bad, just curious :)

**Comment by jaskooo** [comment]
SCClassregisterItem was moved to main class because it is common for two different categories.
I want to move registering classes to a separate class though.

**Comment by lifcio** [comment]
Makes sense.

Overall comment after the changes - looks good, but I don't understand why in some places you `- (void)handleError:(NSError *)error forCompletionBlock:(SCCompletionBlock)completionBlock` and in some you don't. Is there a logic behind it?

E.g. [here](https://github.com/Syncano/syncano-ios/pull/32/files#diff-404c07fc5bc7ca113800decb6ac57d31R40) you use the method, but [here](https://github.com/Syncano/syncano-ios/pull/32/files#diff-404c07fc5bc7ca113800decb6ac57d31R106) you don't. Should all of them use it?

**Pull request merged**